filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -885,19 +885,18 @@ bool VulkanDriver::isWorkaroundNeeded(Workaround workaround) {
 }
 
 FeatureLevel VulkanDriver::getFeatureLevel() {
-    const auto supportedSamplerCount = std::min(
-            mContext.physicalDeviceProperties.limits.maxDescriptorSetSamplers,
-            mContext.physicalDeviceProperties.limits.maxDescriptorSetSampledImages);
-
-    const bool imageCubeArray = (bool)mContext.physicalDeviceFeatures.imageCubeArray;
+    const VkPhysicalDeviceLimits& limits = mContext.physicalDeviceProperties.limits;
+    if (!mContext.physicalDeviceFeatures.imageCubeArray) {
+        return FeatureLevel::FEATURE_LEVEL_1;
+    }
 
-    if (imageCubeArray) {
-        if (supportedSamplerCount >= 31) {
-            return FeatureLevel::FEATURE_LEVEL_3;
-        }
+    const auto& fl2 = FEATURE_LEVEL_CAPS[+FeatureLevel::FEATURE_LEVEL_2];
+    if (fl2.MAX_VERTEX_SAMPLER_COUNT > limits.maxPerStageDescriptorSamplers ||
+        fl2.MAX_FRAGMENT_SAMPLER_COUNT > limits.maxPerStageDescriptorSamplers) {
         return FeatureLevel::FEATURE_LEVEL_2;
     }
-    return FeatureLevel::FEATURE_LEVEL_1;
+
+    return FeatureLevel::FEATURE_LEVEL_3;
```

Comment:
I don't think the condition for fl3 is correct. It never checks against the fl3 values.

Reply:
Doh!  Fixed.